android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -609,9 +615,12 @@ private void downloadKMP(String scheme) {
           FileUtils.getFilename(url);
         }
 
-        // Parse the url for the BCP 47 language ID
-        Uri uri = Uri.parse(url);
-        String languageID = uri.getQueryParameter(KMKeyboardDownloaderActivity.KMKey_BCP47);
+        // Parse data for the BCP 47 language ID
+        String languageID = data.getQueryParameter(KMKeyboardDownloaderActivity.KMKey_BCP47);
```

Comment:
If `bcp47` (and `tag`) are not specified, will this work? Just checking that the parameter is optional.

Reply:
yep, `getQueryParameter()` returns `null if the parameter isn't found.